### DISCUSSION TITLE
Project Name: 'UniDi' or...

### DISCUSSION THREAD

**Discussion by Mathijs-Bakker** [UniDi/UniDi#1] [General]
## Introduction
**Zenject** and also **Extenject** are claimed and copyrighted by Modest Tree. 
However both names and projects are under MIT license,
it is advisable to use an other name. And now is the time for that change.

I came up with the name **UniDi** (without much thought),
but it is still debatable.

**So if anyone comes with a better idea,
please share!**

### A composition of three words:
- Unity, the primary target,
- Universal, you can use it as a standalone library for .NET
- DI, Dependency Injection

### Short:
```cs
using UnityEngine;
using UniDi;
``` 

### Not taken:
- No other repository and organization is has a name like this.
- There's no asset in the Unity Asset Store who has a similar name

> Note: 
> There's a company that is named: UniDiCom. 
> Which operates on: https://unidi.com

## Have a better one?
**So if someone has a better idea for a name,
we change it immediately (project and GitHub organization).**

**Comment by tsgcpp** [comment]
"UniDi" sounds good to me.

I was just wondering "i" of "Di" would be capital letter ("UniDI").
However this is a trivial matter.

**Reply by gngriffiths** [reply]
Funny, I had the same thought. UniDI makes more sense. But it's trivial!

**Reply by Mathijs-Bakker** [reply]
Initially I had the same thought about using `UniDI` over `UniDi`.
After checking .Nets naming conventions for namespaces (and sanity check) -  and we have the `DiContainer` internally...
I decided to go for pascal case.

```csharp
using System.Web.UI;
using System.Xml;
using UniDi;
```

For me it reads better (code wise).

**However:**
For visual identity/communication... UniDI, Uni-DI, UNIDI, UniDi

'UniDI/UniDI'
`UniDI/Signals'

That's an easy change, if it needs to be done.

**Reply by tsgcpp** [reply]
Understood.

It makes sense for me!

> After checking .Nets naming conventions for namespaces (and sanity check) - and we have the DiContainer internally...
